src/main/java/com/adobe/cq/testing/client/TemplateEditorManagerClient.java

```diff
@@ -0,0 +1,258 @@
+
+    private static final String DEFAULT_CONTAINER_RESOURCE_TYPE = "wcm/foundation/components/responsivegrid";
+
+    private static final String PROP_JCR_TITLE = "./jcr:title";
+    private static final String PARAM_DESCRIPTION = "description";
+    private static final String PARAM_PARENT_PATH = "_parentPath_";
+
+    public TemplateEditorManagerClient(final CloseableHttpClient http, final SlingClientConfig config) throws ClientException {
+        super(http, config);
+    }
+
+    public TemplateEditorManagerClient(final URI serverUrl, final String user, final String password) throws ClientException {
+        super(serverUrl, user, password);
+    }
+
+    /**
+     * Creates a default page template of type HTML5
+     * @param configPath path of config to be used to create a page template
+     * @param title title of the template
+     * @param description description of the template
+     * @return created template path
+     * @throws ClientException
+     */
+    public String createDefaultTemplate(final String configPath, final String title, final String description) throws ClientException {
+        return createTemplate(configPath, DEFAULT_TEMPLATE_TYPE, title, description);
+    }
+
+    /**
+     * Creates a page template with given template type
+     * @param configPath path of config to be used to create a page template
+     * @param templateType type of the template
+     * @param title title of the template
+     * @param description description of the template
+     * @return created template path
+     * @throws ClientException
+     */
+    public String createTemplate(final String configPath, final String templateType, final String title, final String description) throws ClientException {
+        FormEntityBuilder formEntry = FormEntityBuilder.create()
+                .addParameter(PARAMETER_CHARSET, CHARSET_UTF8)
+                .addParameter(CSRFUtils.PARAM_CSRF_TOKEN, CSRFUtils.createCSRFToken(this))
+                .addParameter(PROP_JCR_TITLE, title)
+                .addParameter(PARAM_DESCRIPTION, description)
+                .addParameter(PARAM_HIDDEN, "true")
+                .addParameter(PARAM_STATUS, "draft")
+                .addParameter(PARAM_PARENT_PATH, configPath)
+                .addParameter(PARAM_TEMPLATE_TYPE, templateType)
+                .addParameter(PARAM_TEMPLATE_TYPE + "@Delete", "");
+
+        String content = doPost(TEMPLATE_CREATE_ENDPOINT, formEntry.build(), HttpStatus.SC_CREATED).getContent();
+        return extractPath(content);
+    }
+
+    /**
+     * Creates default layout container (responsivegrid) on the provided location of the provided template
+     *
+     * @param templatePath: path of the template the layout container should be created, i.e. /conf/myConfig/settings/wcm/templates/myTemplate
+     * @param location: the location the layout container should be created, i.e. /structure/jcr:content/root/
```

Comment:
this parameter is somehow confusing. Is it relative to the template root? Does it have to exist already?

Reply:
yes and with default template structure a root is already present